### PR TITLE
Microsoft Defender for Endpoint deprecation

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-12304)

## Description
Deprecating multiple commands in the `Microsoft Defender for Endpoint` integration, and the `Microsoft Defender for Endpoint Alerts` collector as well. 

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 1]
```diff
@@ -133,6 +133,38 @@ Please add the following permissions to the app registration. Choose application
...
+
+Note: Following [this](https://learn.microsoft.com/en-us/defender-endpoint/configure-siem) announcement by Microsoft about migrating from the deprecated (and possibly end-of-life) SIEM API to the Graph API, we are deprecating the following:
+
+- **14 commands**
+- **Fetch-incidents functionality**
+
+### Replacement Options:
+- Some commands have direct replacements in the **Microsoft Graph Security** integration.
+- Others do not have exact replacements but offer alternatives that return similar data.
+- A few commands have no available replacements:
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:146]:
```suggestion
- A few commands have no available replacements.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 2]
```diff
@@ -133,6 +133,38 @@ Please add the following permissions to the app registration. Choose application
...
+## Deprecation Details Table
+
+
+| **Command**                     | **Replacement**                                                                                                                                               |
+|----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Fetch incidents`    | Use the `Fetch incidents` in the `Microsoft Graph Security` integration instead.                                                                                                                                    |
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:152]:
```suggestion
| `Fetch incidents`    | Use `Fetch incidents` in the `Microsoft Graph Security` integration instead.                                                                                                                                    |
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_18_0.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+- Command ***microsoft-atp-update-alert*** is deprecated. Use the 'msg-update-alert' command in the 'Microsoft Graph Security' integration instead.
+
+- Command ***microsoft-atp-get-file-alerts*** is deprecated. No available replacement.
+
+- Command ***microsoft-atp-advanced-hunting*** is deprecated. Use the 'msg-advanced-hunting' command in the 'Microsoft Graph Security' integration instead.
+
+- Command ***microsoft-atp-get-alert-related-user*** is deprecated. An alternative is to use the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration, which can retrieve `userAccount` information as part of the alert details.
+
+- Command ***microsoft-atp-get-alert-related-ips*** is deprecated. An alternative is to use the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration, which can retrieve `userAccount` information as part of the alert details.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_18_0.md:36]:
```suggestion
- ***microsoft-atp-get-alert-related-files*** is deprecated. An alternative is the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration, which can retrieve `fileDetails` as part of the alert details.
  
- ***microsoft-atp-get-alert-related-domains*** is deprecated.  An alternative is to use the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration to retrieve `DomainName` as part of the alert details.
  
- ***microsoft-atp-list-alerts*** is deprecated. Use the 'msg-search-alerts' command in the 'Microsoft Graph Security' integration instead.

- ***microsoft-atp-create-alert*** is deprecated. No available replacement.

-  ***microsoft-atp-get-ip-alerts*** is deprecated. No available replacement.

- ***microsoft-atp-get-user-alerts*** is deprecated. No available replacement.

- ***microsoft-atp-get-alert-by-id*** is deprecated. Use 'msg-get-alert-details' in the 'Microsoft Graph Security' integration instead.

- ***microsoft-atp-get-machine-alerts*** is deprecated. No available replacement.

- ***microsoft-atp-get-domain-alerts*** is deprecated. No available replacement.

- ***microsoft-atp-update-alert*** is deprecated. Use the 'msg-update-alert' command in the 'Microsoft Graph Security' integration instead.

- ***microsoft-atp-get-file-alerts*** is deprecated. No available replacement.

-  ***microsoft-atp-advanced-hunting*** is deprecated. Use the 'msg-advanced-hunting' command in the 'Microsoft Graph Security' integration instead.

- ***microsoft-atp-get-alert-related-user*** is deprecated. An alternative is to use the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration, which can retrieve `userAccount` information as part of the alert details.

-  ***microsoft-atp-get-alert-related-ips*** is deprecated. An alternative is to use the 'msg-get-alert-details' command in the 'Microsoft Graph Security' integration, which can retrieve `userAccount` information as part of the alert details.
```

**Comment by ShirleyDenkberg** [comment]
@JasBeilin Doc review completed.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/Microsoft365DefenderEventCollector/README.md [hunk 4]
```diff
@@ -1,5 +1,12 @@
-Microsoft Defender for Endpoint Alerts integration for Cortex XSIAM.
+Microsoft Defender for Endpoint Alerts integration for Cortex XSIAM (Deprecated).
 
+## Deprecation Announcement
+Following [this](https://learn.microsoft.com/en-us/defender-endpoint/configure-siem) announcement by Microsoft about migrating from the deprecated SIEM API to the Graph API, this Event Collector is now deprecated.
+
+### Replacement Option:
+Use the XSIAM `Office 365` Data Source, select `Microsoft Graph API` -> `Alerts`, and choose the `Use Microsoft Graph API V2` option.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/Microsoft365DefenderEventCollector/README.md:7]:
```suggestion
In XSIAM `Office 365` Data Source, click `Microsoft Graph API` > `Alerts`, and select `Use Microsoft Graph API V2`.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_description.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+### Deprecation Note: 
+`Fetch incidents` is deprecated. Use the `Microsoft Graph Security `integration instead.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection_description.md:2]:
```suggestion
`Fetch incidents` is deprecated. Use the `Microsoft Graph Security` integration instead.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 6]
```diff
@@ -3,6 +3,21 @@
...
+
+- **14 commands**
+- **Fetch-incidents functionality**
+
+### Replacement Options:
+- Some commands have direct replacements in the **Microsoft Graph Security** integration.
+- Others do not have exact replacements but offer alternatives that return similar data.
+- A few commands have no available replacements.
+Below, you'll find details on the deprecated commands and their replacements or alternatives.
+[Go to details](#deprecation-details)
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:19]:
```suggestion
[See Deprecation Details](#deprecation-details) to find details on the deprecated commands and their replacements or alternatives.

```

**Comment by ShirleyDenkberg** [comment]
@RosenbergYehuda I reviewed the files again.  Made some small changes.

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**